### ISSUE TITLE
Enhancements for the login component

### ISSUE THREAD

**Issue by mlopezFC** [vaadin/flow-components#5494]
### Describe your motivation

Usually I would like to modify the contents of the login component, so I can show other fields besides user and password. Sometimes I would like to provide a combobox for selecting a branch or things like that. I also would like to be able to create other screens based on the same layout, usually public facing views like the registration view or a view for updating the password before logging in to the application.
Another example might be to be able to show the user the possibility to accept terms and conditions before entering the app, etc.
Another related requirement is that I would like to replace the header or the background with plain images instead of text.
So, in summary, I would like the component to be more "flexible"

### Describe the solution you'd like

It would be great to support the following features:
- Provide a layout of the login view that can be used to create views with the same look and feel than the login view, but with different contents
- Customize the login so the contents of the header, form and footer (place were the forgot password is placed) can be replaced with something else

### Describe alternatives you've considered

I've created [an addon](https://addonsv24.flowingcode.com/extended-login) for this, but I think that maybe these features can be included in the core component in the future.

**Comment by knoobie** [comment]
Already coming in 24.2 https://github.com/vaadin/platform/issues/4491

**Comment by web-padawan** [comment]
> I would like to modify the contents of the login component, so I can show other fields besides user and password.

This was implemented in https://github.com/vaadin/flow-components/pull/5390 as part of the 24.2 scope as linked above.

Note, replacing the existing username and password fields with something else is not currently supported.

**Comment by yuriy-fix** [comment]
Dear @mlopezFC, could you please provide a screenshot / mock of the UI you are trying to build?

**Comment by mlopezFC** [comment]
Suppose I would like to do something like this:

![image](https://github.com/vaadin/flow-components/assets/28312758/fd3244b7-f256-41a9-b200-86d237b17d11)

I don't want to create a login view from zero, I like that the login view is responsive and everything.

There are other examples, I would like to completely change what's inside the `<vaadin-form-layout/>`
